Geometry_Engine/Compute/BooleanIntersection.cs

```diff
@@ -94,6 +104,11 @@ public static Line BooleanIntersection(this Line line, Line refLine, double tole
 
         /***************************************************/
 
+        [Description("Computes the boolean intersection of a line and a collection of other lines, e.g. all overlaps of the first line and the reference lines and returns a new collection of lines matching the overlap.")]
+        [Input("line", "First line to intersect.")]
+        [Input("refLines", "A collection of reference lines to intersect with the first line.")]
+        [Input("tolerance", "Tolerance to be used in the method.", typeof(Length))]
+        [Output("inter", "The collection of lines corresponding to the overlaps of the first line and any of the reference lines.")]
         public static List<Line> BooleanIntersection(this Line line, List<Line> refLines, double tolerance = Tolerance.Distance)
```

Comment:
THat would require checking analytics as well as message for deleted etc. All these descs already made this PR scope creep quite a bit. Prefer to not touch it.

Reply:
Ok, second thought on that: can imagine this being an optimisation for the case when one needs to intersect a single line with a collection of other lines: instead of doing the intersections and then summing them, one can do the union first and then intersect. Possibly more efficient, plus easier to call this method than call `foreach` every time one needs this feature.